x-pack/plugins/security_solution/public/common/utils/global_query_string/index.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as H from 'history';
+import { parse, ParsedQuery, stringify } from 'query-string';
+import { useEffect, useMemo } from 'react';
+
+import { url } from '@kbn/kibana-utils-plugin/public';
+import { isEmpty, pickBy } from 'lodash/fp';
+import { useHistory } from 'react-router-dom';
+import {
+  decodeRisonUrlState,
+  encodeRisonUrlState,
+  getParamFromQueryString,
+  getQueryStringFromLocation,
+} from '../../components/url_state/helpers';
+import { useShallowEqualSelector } from '../../hooks/use_selector';
+import { getStore } from '../../store';
+import { globalUrlParamActions, globalUrlParamSelectors } from '../../store/global_url_param';
+import { useRouteSpy } from '../route/use_route_spy';
+import { getLinkInfo } from '../../links';
+import { SecurityPageName } from '../../../app/types';
+
+interface RegisterUrlParams<State> {
+  urlParamKey: string;
+}
+
+/**
+ * Adds UrlParamKey and the initial value to redux store.
+ */
+export const registerUrlParam = <State>({
+  urlParamKey,
+}: RegisterUrlParams<State>): {
+  state: State | null;
+  deregister: () => void;
+} => {
+  const initialValue = getParamFromQueryString(
+    getQueryStringFromLocation(window.location.search),
+    urlParamKey
+  );
+
+  const store = getStore();
+  store?.dispatch(
```

Comment:
wondering if this should be in a `useEffect`?

Reply:
`registerUrlParam` isn't a hook. I designed it like that because I hoped I could call it from an outside react context. But that is not the case so far. So, I wonder if I should create the `useRegisterUrlParam` hook.